### ISSUE TITLE
Cilium helmrelease max retries exhausted on larger clusters

### ISSUE THREAD

**Issue by vxav** [giantswarm/roadmap#3664]
[Source](https://gigantic.slack.com/archives/C02HLSDH3DZ/p1725441676257269)

It happens sometime when cluster upgrades but cilium is not healthy yet, so it exhausts the retries count and gets stuck in that state.. You can just bump the retries in the HelmRelease and it will unblock it

**Comment by vxav** [comment]
https://gigantic.slack.com/archives/C02EVLE9W/p1731947048179289

**Comment by pipo02mix** [comment]
Another one -> https://gigantic.slack.com/archives/C01BYMF6RN0/p1732707159811019

**Comment by gawertm** [comment]
@yulianedyalkova I believe this is more a topic for Tenet, as CAPA has same issues

(cc @alex-dabija, as Yulia is out this week)

**Comment by alex-dabija** [comment]
> [@yulianedyalkova](https://github.com/yulianedyalkova) I believe this is more a topic for Tenet, as CAPA has same issues
> 
> (cc [@alex-dabija](https://github.com/alex-dabija), as Yulia is out this week)

Actually, Cilium is owned by @giantswarm/team-cabbage . @weatherhog could team Cabbage investigate and fix the issue?

**Comment by vxav** [comment]
It isn't an issue with Cilium though, but how it is installed in the clusters and how the helm controller behaves.

**Comment by weatherhog** [comment]
@alex-dabija I agree with @vxav this is not an cilium issue, Cilium is working fine but during cluster upgrades it exceedes the configured max retries and then needs manual intervention, the short fixed mentioned  to increase the retry number helps straight away.
I would say it is either a cluster upgrade issue and or a helm controller issue. 

From what I have seen it also only happens on clusters with larger amounts of nodes, but maybe this also happens on smaller clusters

**Comment by ubergesundheit** [comment]
As a band-aid increasing install and upgrade retries is being worked on in https://github.com/giantswarm/cluster/pull/394

**Comment by alex-dabija** [comment]
> [@alex-dabija](https://github.com/alex-dabija) I agree with [@vxav](https://github.com/vxav) this is not an cilium issue, Cilium is working fine but during cluster upgrades it exceedes the configured max retries and then needs manual intervention, the short fixed mentioned to increase the retry number helps straight away. I would say it is either a cluster upgrade issue and or a helm controller issue.

Yes, I understand, but I think that the team owning the app also owns the app configuration together with helm release CR. The number of retries are part of the configuration (in my opinion). That was my reasoning for asking Cabbage to take a look.

> From what I have seen it also only happens on clusters with larger amounts of nodes, but maybe this also happens on smaller clusters

Might be. Increasing the number of retries should have almost no negative impact on small clusters.



I think @ubergesundheit's [PR](https://github.com/giantswarm/cluster/pull/394) goes in the right direction because:
* it's not clear what the root cause is for why the cilium helm release exhausted its retries (at least not from the linked Slack threads);
* doing more now would require an additional investigation;
* in my opinion increasing the number of retries reduces the priority of the issue to low.


@ubergesundheit is there a reason why the PR is still a draft?

**Comment by ubergesundheit** [comment]
> is there a reason why the PR is still a draft?

[e2e tests in cluster-aws](https://github.com/giantswarm/cluster-aws/pull/941) which I still need to check out

**Comment by vxav** [comment]
I think I would increase it to 100 or so. I had set it to 50 at some point in the provider chart and it didn't really help in the end.

**Comment by AverageMarcus** [comment]
👋  So.... my thoughts on this - don't make it configurable, instead change the value to `-1` which indicated unlimited retries.

**Comment by weatherhog** [comment]
change was made, should be solved for the future.
